controllers/machinehealthcheck_controller.go

```diff
@@ -192,3 +260,134 @@ func (r *MachineHealthCheckReconciler) clusterToMachineHealthCheck(o handler.Map
 	}
 	return requests
 }
+
+// machineToMachineHealthCheck maps events from Machine objects to
+// MachineHealthCheck objects that monitor the given machine
+func (r *MachineHealthCheckReconciler) machineToMachineHealthCheck(o handler.MapObject) []reconcile.Request {
+	m, ok := o.Object.(*clusterv1.Machine)
+	if !ok {
+		r.Log.Error(errors.New("incorrect type"), "expected a Machine", "type", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	mhcList := &clusterv1.MachineHealthCheckList{}
+	if err := r.Client.List(
+		context.Background(),
+		mhcList,
+		&client.ListOptions{Namespace: m.Namespace},
+		client.MatchingFields{mhcClusterNameIndex: m.Spec.ClusterName},
+	); err != nil {
+		r.Log.Error(err, "Unable to list MachineHealthChecks", "machine", m.Name, "namespace", m.Namespace)
+		return nil
+	}
+
+	var requests []reconcile.Request
+	for k := range mhcList.Items {
```

Comment:
given that we're not paginating the list could not have all the results we need, although this seems unlikely for now, I'd add at least a TODO, wdyt?

Reply:
This is coming from the cache. I'm fairly certain we only need to paginate if we're doing live reads. Right?

Reply:
not 100% sure, what if there are more items? we should be able to test this one easily

Reply:
Follow up from https://kubernetes.slack.com/archives/C0EG7JC6T/p1582925668071800: the apiserver does not force paging on clients, clients must request paging by setting options.Limit > 0. And the controller-runtime informer-based caching client we get from the `Manager` does a full `List()` without pagination.